src/main/java/com/example/validation/validators/EmailValidator.java

```diff
@@ -0,0 +1,34 @@
+package com.example.validation.validators;
+
+import com.example.service.AuthenticationService;
+import com.example.validation.Email;
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+import java.util.regex.Pattern;
+import lombok.RequiredArgsConstructor;
+
+@RequiredArgsConstructor
```

Comment:
```suggestion
```

Reply:
done

Reply:
required args constructor works only for fields, marked with `final`, but you dont have such fields. 
either mark `UserRepository` with `final`, or write custom constructor, or use `@AllArgsConstructor` instead